### PR TITLE
ElasticSearch Handler support

### PR DESCRIPTION
**Pull request by imreFitos**
New ElasticSearchHandler that is compatible with the latest ES2.0 and uses the Elastic Beats format.

### CODE REVIEW

**Comment by shawnbutts** [comment]
I don't see the need to rename the ES collector for this?

**Comment by imreFitos** [comment]
The handler uses the elasticsearch package and the collector's name clobbers the path :( So it won't work unless the collector's filename is different. 

The collector class is still called the same so no configuration has to change.

File: src/diamond/handler/elasticsearchHandler.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        # attempt to connect
+        try:
+            url_list = []
+            for host in self.config['host_urls'].split(','):
+                url_list.append(host)
+            self.es = Elasticsearch(url_list)
+        except Exception as exc:
+            self.log.error("ElasticSearchHandler start failed: {}".format(exc))
+            self.enabled = False
+        # save options into vars
```
Comment (jaingaurav) [inline src/diamond/handler/elasticsearchHandler.py]: Please spread out your code a little to make it easier to read. A newline above here might help.

File: src/diamond/handler/elasticsearchHandler.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        if not self.enabled:
+            return
+        if len(self.queue) == 0:
+            return
+        actions = []
+        action = None
+        current_timestamp = 0
+        current_collector = ''
+        current_host = ''
+        while len(self.queue) > 0:
```
Comment (jaingaurav) [inline src/diamond/handler/elasticsearchHandler.py]: Newline above here

File: src/diamond/handler/elasticsearchHandler.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            self.flush_time = time.time()
+            self.flush()
+
+    def flush(self):
+        """Write all metrics to ES server"""
+        if not self.enabled:
+            return
+        if len(self.queue) == 0:
+            return
+        actions = []
```
Comment (jaingaurav) [inline src/diamond/handler/elasticsearchHandler.py]: Newline

**Comment by jaingaurav** [comment]
Don't have too much to add here aside from style suggestions. Could you please add some newlines/documentation to make the code more readable. Also, it would be best to also fix your commit history to have 2 distinct commits, one to move the collector and the other to add the handler.

**Comment by imreACTmd** [comment]
All right, updated the commit history and fixed up the formatting according to PEP8, let me know what you think!

**Comment by imreACTmd** [comment]
Any comments?

**Comment by stuartbfox** [comment]
My comment is yes please! With grafana v2.5 now supporting native ES searching, it will greatly simplify things if we can output directly to ES

**Comment by imreACTmd** [comment]
Hi, checking in again, is there something wrong with the commit?

**Comment by jaingaurav** [comment]
Hi @imreACTmd looking at the commit history it doesn't seem like you have been able to squash it down to 2 commits as requested. Let me know if you need help with that.

**Comment by jaingaurav** [comment]
Again, the reason for it to be as 2 commits as that it makes it a little bit easier to differentiate between the file move and the handler addition

**Comment by imreFitos** [comment]
@jaingaurav yes please, I thought I managed to redo the commits but I obviously didn't succeed.

**Comment by jaingaurav** [comment]
@imreACTmd @imreFitos http://stackoverflow.com/questions/9790448/how-to-update-a-pull-request should help you out.

**Comment by lpic10** [comment]
Is someone working on this?

**Comment by imreFitos** [comment]
I moved to Sumo Logic and gave up on this.

**Pull request closed without merging**